elytron-oidc-client/src/main/java/org/wildfly/extension/elytron/oidc/ElytronOidcSubsystemTransformers.java

```diff
@@ -52,6 +52,7 @@ public void registerTransformers(SubsystemTransformerRegistration registration)
 
     private static void from3(ChainedTransformationDescriptionBuilder chainedBuilder) {
         ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_3_0_0, VERSION_2_0_0);
+        builder.rejectChildResource(PathElement.pathElement(SECURE_SERVER));
```

Comment:
When transforming to an older version, if the new `scope` attribute is defined for a `secure-deployment`, we need to reject it. Otherwise, if it's not defined, we can just discard it. Take a look here to see how these checks were done when we added a `trusted-virtual-security-domains` attribute to the `security-domain` resource in the Elytron subsystem:

https://github.com/wildfly/wildfly-core/blob/main/elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java#L189-L192

Reply:
We need to do the same for the `secure-server` resource as well, i.e., if the new `scope` attribute is defined for a `secure-server`, we also need to reject it. Otherwise, if it's not defined, we can just discard it.